changelog/env-d-compiler.dd

```diff
@@ -0,0 +1,5 @@
+Use DC environment variable as default D compiler
+
+When DUB recursively invokes itself, it sets up the DC
+env var with the compiler it is using. The `callee` dub should
+use that compiler if it is not overriden with "--compiler". 
```

Comment:
```suggestion
dub now respects the `DC` environment variable, meaning that `DC=ldc2 dub build` will behave as `dub build --compiler=ldc2`. 
In case both are supplied, the `--compiler` switch still has priority.
Note that when DUB recursively invokes itself, for example in `preGenerateCommands`,
it sets the `DC` variable to the compiler it is using, meaning that nested dub invocation will now use the same compiler.
```

I think we should focus more on the general case than to what brought this issue.
It's really a bug that `DC` is not respected in the general case.